rfcs/0006-gtfs-canonical-routes.md

```diff
@@ -0,0 +1,108 @@
+  [docs](https://github.com/mbta/gtfs-documentation/blob/master/reference/gtfs.md) should be updated
+  with this new kind of value.
+- **route_patterns.txt** includes new route patterns for all routes/directions we want to hardcode.
+  `route_pattern_typicality` should be `1`, and `route_pattern_sort_order` should be a less than the
+  value for all other route patterns for that given route/direction. For routes with branches,
+  you'll need a pattern for each route/direction/branch (eg, for Providence/Stoughton Line, you
+  should have two new route patterns in the outbound direction).
+- **shapes.txt** can probably use the existing shapes.
+- **trips.txt** includes one new trip for each new route pattern. It should reference aforementioned
+  route pattern and shapes. It should reference the new `service_id` created above. Each of these
+  trips should be the `representative_trip_id` for their respective route patterns.
+- **stop_times.txt** includes all stops that trains in the route/direction passes through, even if
+  they are temporarily closed. Increment the arrival/departure times by one minute for each
+  subsequent stop. For a sample, see the attached GTFS file, the output of [✓ 🧪 🚝 Canonical
+  trips/patterns experiment](https://app.asana.com/0/881264583703207/1200210504369250), which
+  trialled this for Green Line D and Lowell Line (only in the outbound direction).
+
+This will be done as a new "step" of the Makefile called `canonical`, which runs between `truncate`
+and `current`. There will be a new a directory, `input/canonical/` to store any static data the step
+uses.
+
+The canonical step will be a Python module that reads in the `truncate` feed, and writes out its own
+feed after transforming it. The transformations are as above, and also will update any current trips
+to use canonical route patterns if relevant, eliminating redundant route patterns.
+
+For Commuter Rail route patterns, the canonical one will specify every stop as non-flag, non-leave
+early stop.
+
+# Drawbacks
+
+This is more complicated than not doing it. Some apps may prefer to treat the "typical" route
+pattern as the one that has current service. For example, in the case of a long-term stop closure,
+perhaps some apps don't want to show the closed stop at all.
+
+# Rationale and alternatives
+
+We could leave the current route patterns and trips alone, and add canonical route patterns with a
+new typicality value of `5`. Their only trips would be their single representative trip. With this
+approach, since there's no interaction with the remaining data at all, we would make the `canonical`
+feed one that goes into the `merged` step. (Though we'd have to figure out how to determine the
+valid feed start and end date.)
+
+# Prior art
+
+Until now we've been doing ad-hoc adjustments to determine the route pattern of routes
+[here](https://github.com/mbta/api/blob/master/apps/state/config/config.exs#L145).
```

Comment:
I don't think this tells the whole story — this is configuration in the API, so it doesn't explain the GTFS side, and it's not directly declaring a "canonical" route pattern for routes.

Reply:
I don't actually _think_ we're doing any special `route_pattern_typicality` overrides in GTFS-creator itself, though we do validate that the output looks as we expect it to: https://github.com/mbta/gtfs_creator/blob/master/gtfs_creator2/validate.py#L450.

Either way, it's also a good reminder to think a little more about how this will be implemented in the API for dotcom and others to use/not use.

Reply:
Actually, we do at least one manual `route_pattern_typicality` override for Commuter Rail: https://github.com/mbta/gtfs_creator/blob/master/gtfs_creator2/commuter_rail/route_patterns.py#L64.